servers/rendering/renderer_rd/forward_mobile/scene_shader_forward_mobile.cpp

```diff
@@ -585,10 +544,23 @@ void SceneShaderForwardMobile::init(RendererStorageRD *p_storage, const String p
 		Vector<String> shader_versions;
 		shader_versions.push_back(""); // SHADER_VERSION_COLOR_PASS
 		shader_versions.push_back("\n#define USE_LIGHTMAP\n"); // SHADER_VERSION_LIGHTMAP_COLOR_PASS
-		shader_versions.push_back("\n#define MODE_RENDER_DEPTH\n"); // !BAS! SHADER_VERSION_SHADOW_PASS, should probably change this to MODE_RENDER_SHADOW because we don't have a depth pass here...
-		shader_versions.push_back("\n#define MODE_RENDER_DEPTH\n#define MODE_DUAL_PARABOLOID\n"); // SHADER_VERSION_DEPTH_PASS_DP (maybe rename to SHADER_VERSION_SHADOW_PASS_DP?)
+		shader_versions.push_back("\n#define MODE_RENDER_DEPTH\n"); // SHADER_VERSION_SHADOW_PASS, should probably change this to MODE_RENDER_SHADOW because we don't have a depth pass here...
+		shader_versions.push_back("\n#define MODE_RENDER_DEPTH\n#define MODE_DUAL_PARABOLOID\n"); // SHADER_VERSION_SHADOW_PASS_DP
 		shader_versions.push_back("\n#define MODE_RENDER_DEPTH\n#define MODE_RENDER_MATERIAL\n"); // SHADER_VERSION_DEPTH_PASS_WITH_MATERIAL
+
+		// multiview versions of our shaders
+		shader_versions.push_back("\n#define USE_MULTIVIEW\n"); // SHADER_VERSION_COLOR_PASS_MULTIVIEW
+		shader_versions.push_back("\n#define USE_MULTIVIEW\n#define USE_LIGHTMAP\n"); // SHADER_VERSION_LIGHTMAP_COLOR_PASS_MULTIVIEW
+		shader_versions.push_back("\n#define USE_MULTIVIEW\n#define MODE_RENDER_DEPTH\n"); // SHADER_VERSION_SHADOW_PASS_MULTIVIEW
+
 		shader.initialize(shader_versions, p_defines);
+
+		bool xr_enabled = GLOBAL_GET("rendering/xr/enabled");
```

Comment:
I would most likely have this bool in a bit more central place, like in the compositor, where you can query whether multiview is enabled.

Reply:
You mean like read it once and keep it as a member variable of the compositor that we can read from the other places?

Reply:
Ok, I ran into a problem with the compositor because one of the places this setting is accessed is in the constructor of `EffectsRD`. This object is constructed when storage is constructed, which in turn is constructed within the compositors constructor, which means we don't have access to our variable yet unless we want to start passing pointers to the constructors. 

@reduz I think I'll keep this one for later once we discuss an alternative

Reply:
Ok, now accessing through the singleton on `RendererCompositorRD`, that did the trick :)